File: experiments/spread_bayesian/baselines/qpots/qpots/TS-EMO/dtlz7.py
import torch
import numpy as np
from botorch.test_functions.multi_objective import DTLZ7


def dtlz7(x, dim):
    X = torch.tensor(x, dtype=torch.float32)

    problem = DTLZ7(int(dim), num_objectives=2)
    result = problem.evaluate_true(X)

    return result.numpy()

File: experiments/spread_bayesian/baselines/qpots/qpots/TS-EMO/dtlz7.m
function y = dtlz7(x)
  dim = size(x, 2);
  module = py.importlib.import_module('dtlz7');

  py_x = py.numpy.array(reshape(x, [], dim));
  py_y = module.dtlz7(py_x, dim);

  y = double(py.array.array('d', py.numpy.nditer(py_y)));
end

  
